parser/wasm_parse/mod.rs

```diff
@@ -1,95 +1,121 @@
 use super::Parse;
 use ir::{self, Id};
-use parity_wasm::elements::{self, Section, Type};
-use std::fmt::Write;
-use traits;
-
-fn serialized_size<T>(t: T) -> Result<u32, traits::Error>
-where
-    T: elements::Serialize,
-    traits::Error: From<<T as elements::Serialize>::Error>,
-{
-    let mut buf = vec![];
-    t.serialize(&mut buf)?;
-    Ok(buf.len() as u32)
+use std::collections::HashMap;
+use wasmparser::SectionWithLimitedItems;
+use wasmparser::{self, Operator, SectionReader, Type};
+
+#[derive(Default)]
+pub struct SectionIndices {
+    type_: Option<usize>,
+    code: Option<usize>,
+    functions: Vec<Id>,
+    tables: Vec<Id>,
+    memories: Vec<Id>,
+    globals: Vec<Id>,
```

Comment:
Ah ok, so these map from index space to the id we have assigned to the parsed thing? Nice :+1: